make_a_plea/management/commands/zendesk.py

```diff
@@ -0,0 +1,234 @@
+    u'custom_fields',
+    u'sharing_agreement_ids',
+    u'via',
+]
+
+# Some may contain various line endings that will be coerced to \n
+NEWLINE_FIELDS = [
+    "description",
+    "subject",
+    "raw_subject",
+]
+
+
+class Command(BaseCommand):
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '--to-csv',
+            action='store_true',
+            dest='to-csv',
+            default=False,
+            help='Export all relevant zendesk items to csv file'),
+        )
+
+    def handle(self, *args, **options):
+        self.load_settings()
+        self.init_zen()
+
+        if options["to-csv"]:
+            self.to_csv()
+
+    def load_settings(self):
+        with open(os.path.expanduser("~/.hmcts-zendesk"), "r") as config_file:
+            self.settings = yaml.load(config_file.read())
+
+    def init_csv(self, f, dialect=csv.excel, encoding="utf-8", **kwds):
+        self.queue = cStringIO.StringIO()
+        self.writer = csv.writer(self.queue, dialect=dialect, **kwds)
+        self.stream = f
+        self.encoder = codecs.getincrementalencoder(encoding)()
+
+    def init_zen(self):
+        self.zendesk = Zendesk(
+            self.settings["url"],
+            self.settings["email"],
+            self.settings["password"])
+
+    def get_all_fieldnames(self):
+        return \
+            SIMPLE_FIELDS + \
+            NEWLINE_FIELDS + \
+            STRUCTURED_FIELDS
+
+    def to_csv(self):
+
+        self.outfile = self.settings["outfile"] \
+            if "outfile" in self.settings \
+            else "zendump.csv"
+        self.batch_size = 100
+        with open(self.outfile, "wb") as f:
+
+            self.init_csv(f)
+            self.writerow(self.get_all_fieldnames())  # CSV Header
+
+            for batch in self.yield_ticket_batches():
+                for ticket in batch:
+                    self.writerow(self.ticket_to_row(ticket))
+
+    def writerow(self, row):
+        self.writer.writerow([s.encode("utf-8") for s in row])
+        data = self.queue.getvalue()
+        data = data.decode("utf-8")
+        data = self.encoder.encode(data)
+        self.stream.write(data)
+        self.queue.truncate(0)
+
+    def ticket_to_row(self, ticket):
+        return \
+            self.get_simple_fields(ticket) + \
+            self.get_newline_fields(ticket) + \
+            self.get_structured_fields(ticket)
+
+    def parse_simple_field(self, field):
+        try:
+            return field.encode("utf-8")
+        except AttributeError:  # ints
+            return str(field).encode("utf-8")
+
+    def get_simple_fields(self, ticket):
+        fields = []
+        for field in SIMPLE_FIELDS:
+            if field in ticket:
+                fields.append(self.parse_simple_field(ticket[field]))
+            else:
+                fields.append("")
+        return fields
+
+    def parse_newline_field(self, field):
+        """Newlines make for a messy csv so coerce them to \n"""
```

Comment:
They make for a messy csv when you look at it in a text editor but if we expecting people look at this in a spreadsheet strange characters may be more confusing.